src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -580,7 +582,35 @@ void CoreWorkerDirectTaskSubmitter::PushNormalTask(
           RAY_CHECK_GE(scheduling_key_entry.num_busy_workers, 1u);
           scheduling_key_entry.num_busy_workers--;
 
+          std::thread::id thread_used = std::this_thread::get_id();
           if (!status.ok() || !is_actor_creation || reply.worker_exiting()) {
+            RAY_LOG(DEBUG) << "Getting error from raylet for task " << task_id;
+            
+            auto callback =
+              [this, status, is_actor, task_id, thread_used, addr, reply](const Status &get_task_result_reply_status,
+                     const rpc::GetTaskResultReply &get_task_result_reply) {
+                std::thread::id callback_thread_id = std::this_thread::get_id();
+                RAY_CHECK(thread_used == callback_thread_id) << "Threading issue. Callback must be called on the original calling thread. Original thread id " << thread_used << " callback thread id " << callback_thread_id;
+                RAY_CHECK(!status.ok());
+                
+                rpc::ErrorType task_error_type = rpc::ErrorType::WORKER_DIED;
+                if (get_task_result_reply_status.ok()) {
+                  task_error_type = get_task_result_reply.failure_cause().error_type();
+                  RAY_LOG(DEBUG) << "Task failure cause " << ray::gcs::RayExceptionToString(get_task_result_reply.failure_cause());
+                } else {
+                  RAY_LOG(DEBUG) << "Failed to fetch task result with status " << get_task_result_reply_status.ToString();
+                  // TODO: It'd be nice to detect node failure here.
```

Comment:
is it TODO in this PR? I think it is pretty easy to add it here. We can also add more metadata like node id

Reply:
do you mean we log it as node failure if the rpc failed?